### ISSUE TITLE
Installing unpatched libraries is allowed?

### ISSUE THREAD

**Issue by mgsloan** [ghcjs/ghcjs#434]
I noticed that GHCJS will happily install new versions of packages which require patches / shims:

```
mgsloan@computer:~$ ghcjs-pkg field directory version 
version: 1.2.2.0
mgsloan@computer:~$ cabal install directory --ghcjs
Resolving dependencies...
Downloading directory-1.2.4.0...
```

This could lead to someone getting runtime errors due to installing an unpatched package.  Seems rather non-ideal, but I'm not sure what to do about it.

**Comment by luite** [comment]
yep this is a problem, and i don't have a solution either. i've mostly been diligently updating packages to the latest version. we could manage a black list in ghcjs, since cabal passes a package-name to the compiler, but it really should be handled at the solver level.

**Comment by tolysz** [comment]
The alternative would be to reboot the `ghcjs` just for this build by using the exact versions from the `stack.yaml`... having a fixed list is a bit tricky, as the packages mentioned might have some different dependencies overtime....
